internal/locate/region_request.go

```diff
@@ -557,11 +557,35 @@ func (state *accessFollower) next(bo *retry.Backoffer, selector *replicaSelector
 	}
 	// If there is no candidate, fallback to the leader.
 	if selector.targetIdx < 0 {
-		if len(state.option.labels) > 0 {
-			logutil.BgLogger().Warn("unable to find stores with given labels")
-		}
 		leader := selector.replicas[state.leaderIdx]
-		if leader.isEpochStale() || leader.isExhausted(1) {
+		leaderInvalid := leader.isEpochStale() || leader.isExhausted(1)
+		if len(state.option.labels) > 0 {
+			var regionID uint64
+			var labels []string
+			var replicaStatus []string
+			if selector.region != nil {
+				regionID = selector.region.GetID()
+			}
+			for _, label := range state.option.labels{
+				labels = append(labels, label.Key + ": " + label.Value)
+			}
+			for _, replica := range selector.replicas {
+				replicaStatus = append(replicaStatus, fmt.Sprintf("peer: %v, store: %v, isEpochStale: %v, IsLabelsMatch: %v, attempts: %v",
+					replica.peer.GetId(),
+					replica.store.storeID,
+					replica.isEpochStale(),
+					replica.store.IsLabelsMatch(state.option.labels),
```

Comment:
Maybe the `isExhausted(1)` result could be printed for each peer too, so we know if the peer has been tried in this `replicaSelector` round. Another choice to reduce log printing is to print the detailed peer status when `epochNotMatchError` is returned, which means we print peer details just one time for one `replicaSelector` round.

Besides, it's better if we could print the reason for the failure retry on each peer, the related information would be collected at the `OnRegionError` call.